backend/de.metas.business/src/main/java/de/metas/invoice/review/InvoiceReview.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * #%L
+ * de.metas.business
+ * %%
+ * Copyright (C) 2023 metas GmbH
+ * %%
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation, either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program. If not, see
+ * <http://www.gnu.org/licenses/gpl-2.0.html>.
+ * #L%
+ */
+
+package de.metas.invoice.review;
+
+import de.metas.invoice.InvoiceId;
+import de.metas.organization.OrgId;
+import lombok.Builder;
+import lombok.NonNull;
+import lombok.Singular;
+import lombok.Value;
+
+import javax.annotation.Nullable;
+import java.util.Map;
+
+@Value
+@Builder(toBuilder = true)
+public class InvoiceReview
+{
+	@Nullable InvoiceReviewId id;
```

Comment:
make it mandatory

Reply:
The problem stemmed from the us using the (anti)pattern of using the same domain object when persisted and before persisting. As such, its ID may or may not be available, hence the @Nullable.

Renamed to `InvoiceReviewCreateUpdateRequest`, dropped the `InvoiceReviewId` and `withExtendedProps(...)` and should now be peachy.